airbyte-ci/connectors/pipelines/pipelines/steps/simple_ci_step.py

```diff
@@ -0,0 +1,115 @@
+from pipelines import main_logger
+from dataclasses import dataclass
+from pathlib import Path
+from pipelines.actions.environments import with_installed_pipx_package, with_python_base
+from pipelines.bases import Step, StepResult
+from pipelines.contexts import PipelineContext
+
+import dagger
+
+from typing import List, Optional
+
+
+@dataclass(frozen=True)
+class MountPath:
+    path: Path
+    optional: bool = False
+
+    def __post_init__(self):
+        if not self.path.exists():
+            message = f"{self.path} does not exist."
+            if self.optional:
+                raise FileNotFoundError(message)
+            else:
+                main_logger.warning(message)
+
+    def __str__(self):
+        return str(self.path)
+
+    @property
+    def is_file(self) -> bool:
+        return self.path.is_file()
+
+
+class SimpleCIStep(Step):
+    def __init__(
+        self,
+        title: str,
+        context: PipelineContext,
+        paths_to_mount: List[MountPath] = [],
+        internal_tools: List[MountPath] = [],
+        secrets: dict[str, dagger.Secret] = {},
+        env_variables: dict[str, str] = {},
+        working_directory: str = "/",
+        command: Optional[List[str]] = None,
+    ):
+        """A simple step that runs a given command in a container.
+
+        Args:
+            title (str): name of the step
+            context (PipelineContext): context of the step
+            paths_to_mount (List[MountPath], optional): directory paths to mount. Defaults to [].
+            internal_tools (List[MountPath], optional): internal tools to install. Defaults to [].
+            secrets (dict[str, dagger.Secret], optional): secrets to add to container. Defaults to {}.
+            env_variables (dict[str, str], optional): env variables to set in container. Defaults to {}.
+            working_directory (str, optional): working director to run the command in. Defaults to "/".
+            command (Optional[List[str]], optional): The default command to run. Defaults to None.
+        """
+        self.title = title
+        super().__init__(context)
+
+        self.paths_to_mount = paths_to_mount
+        self.working_directory = working_directory
+        self.internal_tools = internal_tools
+        self.secrets = secrets
+        self.env_variables = env_variables
+        self.command = command
+
+    def _mount_paths(self, container: dagger.Container) -> dagger.Container:
+        for path_to_mount in self.paths_to_mount:
+            path_string = str(path_to_mount)
+            destination_path = f"/{path_string}"
```

Comment:
Can probably leave this out until it's required - but I know a lot of volume mounting configuration usually allows to configure a separate destination path. if it's easy, could influence a different data type for mounts